Metrolist crash

### Checklist - Required

- [x]  I am able to reproduce the bug with the [latest debug version](https://github.com/MetrolistGroup/Metrolist/actions/workflows/build.yml?query=branch%3Amain).
- [x]  I've checked that there is NO open or closed issue about this bug. (Duplicate issues will be closed immediately without response)
- [x]  This issue contains only ONE bug. PLEASE check pinned issues before opening a new one.
- [x]  The title of this issue accurately describes the bug.
- [x]  The entire issue report is written/translated in English.

### Steps to reproduce the bug

Open the app 

### Expected behavior

It shouldn't happen. 

### Actual behavior

Sometimes the app crashes randomly with this error:

```log
Metrolist Crash Report
==================================================

Manufacturer: Google
Device: Pixel 7a
Android version: 16 (SDK 36)
App version: 13.3.0 (143)

==================================================
Stacktrace:
==================================================

java.lang.RuntimeException: Unable to destroy activity {com.metrolist.music/com.metrolist.music.MainActivityAlias}
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6567)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:6599)
	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:52)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:63)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:169)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:101)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2987)
	at android.os.Handler.dispatchMessageImpl(Handler.java:142)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at android.os.Looper.loopOnce(Looper.java:269)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9333)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:929)
Caused by: java.lang.IllegalArgumentException: Service not registered: com.metrolist.music.MainActivity$serviceConnection$1@c7b3646
	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:2222)
	at android.app.ContextImpl.unbindService(ContextImpl.java:2373)
	at android.content.ContextWrapper.unbindService(ContextWrapper.java:973)
	at com.metrolist.music.MainActivity.onDestroy(r8-map-id-e7962ebb1e32a59ddc82e93d9bd8eb9a674ad71011bfeb1137f59c00ea02fdde:67)
	at android.app.Activity.performDestroy(Activity.java:9612)
	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1578)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6554)
	... 15 more
```

### Screenshots/Screen recordings

Nothing 

### Logs

```log
Metrolist Crash Report
==================================================

Manufacturer: Google
Device: Pixel 7a
Android version: 16 (SDK 36)
App version: 13.3.0 (143)

==================================================
Stacktrace:
==================================================

java.lang.RuntimeException: Unable to destroy activity {com.metrolist.music/com.metrolist.music.MainActivityAlias}
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6567)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:6599)
	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:52)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:63)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:169)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:101)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2987)
	at android.os.Handler.dispatchMessageImpl(Handler.java:142)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at android.os.Looper.loopOnce(Looper.java:269)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9333)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:929)
Caused by: java.lang.IllegalArgumentException: Service not registered: com.metrolist.music.MainActivity$serviceConnection$1@c7b3646
	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:2222)
	at android.app.ContextImpl.unbindService(ContextImpl.java:2373)
	at android.content.ContextWrapper.unbindService(ContextWrapper.java:973)
	at com.metrolist.music.MainActivity.onDestroy(r8-map-id-e7962ebb1e32a59ddc82e93d9bd8eb9a674ad71011bfeb1137f59c00ea02fdde:67)
	at android.app.Activity.performDestroy(Activity.java:9612)
	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1578)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6554)
	... 15 more
```

### Metrolist version

Latest

### Android version

Android 16

### Additional information

Nothing

--- a/app/src/main/kotlin/com/metrolist/music/MainActivity.kt
+++ b/app/src/main/kotlin/com/metrolist/music/MainActivity.kt
@@ -270,6 +270,20 @@ class MainActivity : ComponentActivity() {
             }
         }
 
+    private fun safeUnbindService(source: String) {
+        if (!isServiceBound) return
+        try {
+            unbindService(serviceConnection)
+        } catch (e: IllegalArgumentException) {
+            Timber.tag("MainActivity").w(e, "Service was not bound when attempting to unbind in $source")
+        } finally {
+            isServiceBound = false
+            listenTogetherManager.setPlayerConnection(null)
+            playerConnection?.dispose()
+            playerConnection = null
+        }
+    }
+
     override fun onStart() {
         super.onStart()
         // Request notification permission on Android 13+
@@ -291,14 +305,7 @@ class MainActivity : ComponentActivity() {
     }
 
     override fun onStop() {
-        if (isServiceBound) {
-            try {
-                unbindService(serviceConnection)
-            } catch (e: IllegalArgumentException) {
-                Timber.tag("MainActivity").w(e, "Service was not bound when attempting to unbind in onStop()")
-            }
-            isServiceBound = false
-        }
+        safeUnbindService("onStop()")
         super.onStop()
     }
 
@@ -310,15 +317,7 @@ class MainActivity : ComponentActivity() {
         ) {
             stopService(Intent(this, MusicService::class.java))
         }
-        if (isServiceBound) {
-            try {
-                unbindService(serviceConnection)
-            } catch (e: IllegalArgumentException) {
-                Timber.tag("MainActivity").w(e, "Service was not bound when attempting to unbind in onDestroy()")
-            }
-            isServiceBound = false
-        }
-        playerConnection = null
+        safeUnbindService("onDestroy()")
     }
 
     override fun onNewIntent(intent: Intent) {

